src/include/simeng/pipeline/DispatchIssueUnit.hh

```diff
@@ -125,6 +124,19 @@ class DispatchIssueUnit {
    * at `dependencyMatrix[type][tag]`. */
   std::vector<std::vector<std::vector<dependencyEntry>>> dependencyMatrix_;
 
+  /** Contains all of the instructions that have a dependency with another
+   * instruction, but is not permitted to have the result forwarded to them
+   * directly. */
+  std::vector<dependencyEntry> dependantInstructions_;
+
+  /** Contains instructions that have had results forwarded to them, but need to
+   * wait for x-cycles to mimic the in hardware latency of said result
+   * forwarding.
+   * Outer Pair = {ticks_ value when latency has elapsed, Inner Pair}.
+   * Inner Pair = {instruction entry, value it has been forwarded}. */
+  std::vector<std::pair<uint64_t, std::pair<dependencyEntry, RegisterValue>>>
```

Comment:
I agree. I have updated the `waitingInstructions_` data structure to be an unordered map : `std::unordered_map<uint64_t, std::vector<std::pair<dependencyEntry, RegisterValue>>>`

This allows us to "fire off" all instructions in the vector for the current tick count. A map was used instead of a queue due the bypass latency can vary depending on which instruction is recieving the result. This means that cycle to release on will not always be equal to or greater than current entries in the map.